grpcbp/server_middlewares.go

```diff
@@ -0,0 +1,77 @@
+package grpcbp
+
+import (
+	"context"
+
+	"github.com/reddit/baseplate.go/ecinterface"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/tracing"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+func InjectServerSpanInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (_ interface{}, err error) {
+		ctx, span := StartSpanFromGRPCContext(ctx, info.FullMethod)
+		defer func() {
+			span.FinishWithOptions(tracing.FinishOptions{
+				Ctx: ctx,
+				Err: err,
+			}.Convert())
+		}()
+		return handler(ctx, req)
+	}
+}
+
+func InjectEdgeContextInterceptor(impl ecinterface.Interface) grpc.UnaryServerInterceptor {
+	if impl == nil {
+		impl = ecinterface.Get()
+	}
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (_ interface{}, err error) {
+		ctx = InitializeEdgeContext(ctx, impl)
+		return handler(ctx, req)
+	}
+}
+
+func InitializeEdgeContext(ctx context.Context, impl ecinterface.Interface) context.Context {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return ctx
+	}
+	values, ok := md[HeaderEdgeRequest]
+	if !ok {
+		return ctx
+	}
+
+	ctx, err := impl.HeaderToContext(ctx, values[0])
+	if err != nil {
+		log.Error("Error while parsing EdgeRequestContext: " + err.Error())
```

Comment:
(I know this is copied from the thriftbp code, that should be fixed as well,)

if we are not going to use a `log.Wrapper` here (which is debatable, as passing in a `log.Wrapper` all the way here just for this line might be considered overkill), then we should at least use zap logger properly:

```
log.C(ctx).Errorw("Error while parsing EdgeRequestContext", "err", err)
```